uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -433,6 +474,34 @@ define([
       }
 
       /**
+       * Pick a specific leave request day type
+       *
+       * @param {string} value - The leave request day type value to match
+       * @return {object}
+       */
+      function getStatusFromValue(value) {
+        var collection = vm.leaveRequestStatuses,
+          key = 'value';
```

Comment:
There is no point defining `collection` either, the function body can be much more terse
```js
function getStatusFromValue(value) {  
  return _.find(vm.leaveRequestStatuses, function (status) {
    return status['value'] === value;
  });
}
```
Also please make sure to give meaningful names to the arguments of the callback (`status`, not a generic `collectionItem`)